bin/oneoff/generate_legacy_survey_summaries.rb

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env ruby
+require_relative '../../dashboard/config/environment'
+require 'pd/survey_pipeline/survey_pipeline_helper.rb'
+
+# for pegasus survey summaries
+include Api::V1::Pd::WorkshopScoreSummarizer
+
+# for jotform survey summaries
+include Pd::SurveyPipeline::Helper
+
+# As we deprecate pegasus- and jotform-based surveys in favour of our in-house Foorm system,
+# we want to snapshot certain summaries that we have shown to facilitators in the past.
+# This script generates those values and records them in a new Pd::LegacySurveySummary model.
+
+def main
+  case ARGV[0]
+  when "csf_intro_post_workshop"
+    snapshot_csf_intro_post_workshop_from_pegasus
+  when "csf_intro_post_workshop_all_workshops"
+    snapshot_csf_intro_post_workshop_from_pegasus_for_all_workshops
+  when "csd_summer_workshops"
+    snapshot_csd_summer_workshops_from_jotform
+  when "csp_summer_workshops"
+    snapshot_csp_summer_workshops_from_jotform
+  else
+    puts "Usage: generate_legacy_survey_summaries.rb"
+    puts "           csf_intro_post_workshop |"
+    puts "           csf_intro_post_workshop_all_workshops |"
+    puts "           csd_summer_workshops |"
+    puts "           csp_summer_workshops"
+  end
+end
+
+def snapshot_csf_intro_post_workshop_from_pegasus
+  course = Pd::Workshop::COURSE_CSF
+  subject = Pd::Workshop::SUBJECT_CSF_101
+
+  # facilitator_id: results
+  facilitators = {}
+
+  # let's get all workshops for CSF Intro, that are ended, and not summer
+  all_completed_csf_info_workshops = Pd::Workshop.where(course: course, subject: subject).in_state(Pd::Workshop::STATE_ENDED).exclude_summer
+
+  # let's find all facilitators who have done these.  we use a hash with ID as key so that we only
+  # record each facilitator once
+  all_completed_csf_info_workshops.each do |w|
+    w.facilitators.each do |f|
+      unless facilitators.key?(f.id)
+        facilitators[f.id] = f
+      end
+    end
+  end
+
+  survey_reports = {}
+
+  facilitators.first(10).each do |id, f|
+    all_their_workshops = Pd::Workshop.facilitated_by(f)
+    all_their_completed_workshops = all_their_workshops.where(course: course, subject: subject).in_state(Pd::Workshop::STATE_ENDED).exclude_summer
+
+    all_their_workshops_for_course = get_score_for_workshops(
```

Comment:
Was going to ask how you test that what this script comes up with matches the summaries people would have seen previously, but I think that's irrelevant since it looks like you're just using an existing function for summarizing scores? Is that right?

Reply:
We are using an existing function, yes.  That said, we will need to verify this seems to be working before it goes live.  If nothing else, I intend to spot-check a sampling of real accounts to make sure the legacy UI and the new UI have matching numbers.